lib/Tools/LinearLayout.cpp

```diff
@@ -918,6 +918,15 @@ LinearLayout LinearLayout::invertAndCompose(const LinearLayout &outer) const {
   return flatComposed.reshapeIns(retInDims).reshapeOuts(retOutDims);
 }
 
+LinearLayout LinearLayout::invert() const {
```

Comment:
@Mogball highly requested feature :P

Reply:
Also, perhaps for now you want to assert that the layout is invertible? Meaning that it has the same total number of inputs and outputs and it's surjective, for example.

Reply:
Yeah, I'll do that

Reply:
Indeed